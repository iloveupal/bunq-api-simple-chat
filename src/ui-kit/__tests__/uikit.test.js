import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Button, ButtonRow, Input, MultipleSelect, Switch } from '..';

describe('UiKit', () => {
    test('Button should match the snapshot', () => {
        const component = renderer.create(
            <Button
                onClick={jest.fn()}
            >
                Test
            </Button>
        );

        expect(component.toJSON()).toMatchSnapshot();

        component.update(
            <Button
                primary
                onClick={jest.fn()}
            >
                Test
            </Button>
        );

        expect(component.toJSON()).toMatchSnapshot();

        component.update(
            <Button
                isDisabled
                onClick={jest.fn()}
            >
                Test
            </Button>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('ButtonRow should match the snapshot', () => {
        const component = renderer.create(
            <ButtonRow>
                <div />
                <div />
            </ButtonRow>
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('Input should match the snapshot', () => {
        const component = renderer.create(
            <Input
                value={'test'}
                onChange={jest.fn()}
            />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('MultipleSelect should match the snapshot', () => {
        const component = renderer.create(
            <MultipleSelect
                options={['1', '2']}
                value={['1']}
                renderItem={({ option, isSelected }) => <div key={option}>{String(isSelected)}</div>}
                onChange={jest.fn()}
            />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    test('MultipleSelect should call onChange when the inner component calls onClick', () => {
        const TestComponent = ({ option, isSelected, onClick }) => (
            <div onClick={onClick} key={option} testId="1"/>
        );

        const onChangeFn = jest.fn();

        const component = renderer.create(
            <MultipleSelect
                options={['1']}
                value={['1']}
                renderItem={TestComponent}
                onChange={onChangeFn}
            />
        );

        component.root.findByProps({ testId: '1' }).props.onClick();

        expect(onChangeFn).toHaveBeenCalled();
    });

    test('MultipleSelect should pass isSelected to selected components', () => {
        const TestComponent = ({ option, isSelected, onClick }) => (
            <div onClick={onClick} key={option} testId={option} isSelected={isSelected} />
        );

        const component = renderer.create(
            <MultipleSelect
                options={['1', '2', '3']}
                value={['1', '2']}
                renderItem={TestComponent}
                onChange={jest.fn()}
            />
        );

        expect(component.root.findByProps({ testId: '1' }).props.isSelected).toBe(true);
        expect(component.root.findByProps({ testId: '2' }).props.isSelected).toBe(true);
        expect(component.root.findByProps({ testId: '3' }).props.isSelected).toBe(false);
    });
});