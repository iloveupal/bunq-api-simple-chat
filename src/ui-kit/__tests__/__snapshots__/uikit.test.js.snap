// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiKit Button should match the snapshot 1`] = `
.c0 {
  height: 40px;
  min-width: 150px;
  padding: 0 15px;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  font-family: sans-serif;
  font-weight: normal;
  cursor: pointer;
  opacity: 1;
}

.c0:hover {
  background-color: #333;
}

<div
  className="c0"
  onClick={[MockFunction]}
>
  Test
</div>
`;

exports[`UiKit Button should match the snapshot 2`] = `
.c0 {
  height: 40px;
  min-width: 150px;
  padding: 0 15px;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  font-family: sans-serif;
  font-weight: bold;
  cursor: pointer;
  opacity: 1;
}

.c0:hover {
  background-color: #333;
}

<div
  className="c0"
  onClick={[MockFunction]}
>
  Test
</div>
`;

exports[`UiKit Button should match the snapshot 3`] = `
.c0 {
  height: 40px;
  min-width: 150px;
  padding: 0 15px;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  font-family: sans-serif;
  font-weight: normal;
  cursor: pointer;
  opacity: 0.5;
}

<div
  className="c0"
  onClick={null}
>
  Test
</div>
`;

exports[`UiKit ButtonRow should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
>
  <div />
  <div />
</div>
`;

exports[`UiKit Input should match the snapshot 1`] = `
.c0 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  height: 40px;
  background-color: #333;
  border-radius: 5px;
  border: none;
  padding: 0 15px;
  font-size: 15px;
  color: white;
}

.c0:focus {
  outline: none;
}

<input
  className="c0"
  onChange={[Function]}
  value="test"
/>
`;

exports[`UiKit MultipleSelect should match the snapshot 1`] = `
.c0 {
  height: 300px;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  overflow-y: scroll;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

<div
  className="c0"
>
  <div>
    true
  </div>
  <div>
    false
  </div>
</div>
`;
